### DISCUSSION TITLE
Document uploads

### DISCUSSION THREAD

**Discussion by rmeekers** [danny-avila/LibreChat#926] [Q&A] [answered]
Hi

Is there a possibility to upload documents (text, pdf, ...) to provide extra context? Or to summarize a doc?

Thanks

**Comment by danny-avila** [comment, marked as answer]
Not yet. It's on the roadmap: https://github.com/users/danny-avila/projects/2?pane=issue&itemId=30598364

I've been working on further securing the app as a lot of people are deploying. It's next on my list.

**Reply by ventz** [reply]
@danny-avila Would it make sense as a simple in between to allow for pdfs/text/docs/etc via the front end, and simply to load the files into the input context as though they are pasted into the input field? With larger and larger input LLM contexts (200K Claude and 128K with OpenAI), it seems this is good enough for most to allow users to interact with files.

We did that locally here as an in between step to RAG, and it turned out to be more than sufficient for ~9K users with a custom AI sandbox (not based on LibreChat)

**Reply by danny-avila** [reply]
@ventz that is planned already

**Reply by ventz** [reply]
Awesome - thank you!
